app/controllers/stability_controller.rb

```diff
@@ -98,6 +98,110 @@ def index
     ]
   end
 
+  # pass ?weeks=x to 
+  def okrs
+    # Render our OKR goals around fastlane stability
+    tools = {
+      cert: 0.89,
+      deliver: 0.7,
+      fastlane: 0.7,
+      frameit: 0.9,
+      gym: 0.76,
+      match: 0.88,
+      pem: 0.88,
+      pilot: 0.7,
+      produce: 0.85,
+      scan: 0.65, # this is lower than 0.7, beacuse failing tests also report a lower success rate
+      screengrab: 0.7,
+      sigh: 0.86,
+      snapshot: 0.85,
+      supply: 0.73
+    }
+
+    actions = {
+      cocoapods: 0.9,
+      increment_build_number: 0.95,
+      slack: 0.95,
+      clear_derived_data: 0.95,
+      set_info_plist_value: 0.95,
+      gradle: 0.8,
+      get_version_number: 0.95,
+      crashlytics: 0.9,
+      get_info_plist_value: 0.95,
+      xcodebuild: 0.9,
+      hockey: 0.94,
+      carthage: 0.9,
+      unlock_keychain: 0.9,
+      update_fastlane: 0.95,
+      update_info_plist: 0.95,
+      reset_git_repo: 0.95,
+      increment_version_number: 0.95,
+      get_build_number: 0.95,
+      xcode_select: 0.95,
+      update_app_identifier: 0.95,
+      update_project_provisioning: 0.95,
+      badge: 0.95,
+      changelog_from_git_commits: 0.95,
+      add_git_tag: 0.95,
+      ensure_git_branch: 0.95,
+      hipchat: 0.95,
+      register_devices: 0.95
+    }
+
+    @okrs_tools = []
+    @okrs_actions = []
+
+    tools.merge(actions).each do |action_name, goal|
+      data_for_action = select_details_info(action_name, (params["weeks"] || 3 * 4).to_i) # defaults to one quarter
```

Comment:
I suspect that this is now making something like 30+ queries per page load, which is probably not great. We should think about whether we want to try to improve it right now.